js/game.js

```diff
@@ -1149,16 +1149,18 @@ function init_socket(args)
 		window.socket.destroy();
 	}
 	$(".disconnected").hide();
-	if(is_sdk && (Cookies.get("windows") || Cookies.get("local_ip") || window.location.host=="advanture.land" || window.location.host=="x.thegame.com")) server_addr="192.168.1.125"; // Cookies.set('windows','1',{expires:12*365});
-	else if(is_sdk)
-	{
-		if(window.location.origin=='http://127.0.0.1/') server_addr="127.0.0.1";
-		else server_addr="0.0.0.0";
-	}
+	// if(is_sdk && (Cookies.get("windows") || Cookies.get("local_ip") || window.location.host=="advanture.land" || window.location.host=="x.thegame.com")) server_addr="192.168.1.125"; // Cookies.set('windows','1',{expires:12*365});
+	// else if(is_sdk)
+	// {
+	// 	server_addr = window.location.hostname
+	// 	console.log('SDK MODE wss server adress:', location.protocol, server_addr, server_port)
+	// }
+
+	add_log("Connecting to the server.");
+	add_log(`${location.protocol} ${server_addr} ${server_port}`);
 	var query=args.secret&&"desktop="+(!is_comm&&1||"")+"&secret="+args.secret||undefined;
 	if(location.protocol=="https:") window.socket=io('wss://'+server_addr+':'+server_port,{secure:true,transports:['websocket'],query:query});
-	else window.socket=io(server_addr+':'+server_port,{transports:['websocket'],query:query});
-	add_log("Connecting to the server.");
+	else window.socket=io('ws://'+server_addr+':'+server_port,{transports:['websocket'],query:query});
```

Comment:
Why does it now explicitely need the protocol but didn't before?

Reply:
To be able to connect via caracAL to the server on http, the protocol was required for a correct websocket / socket.io connection to be established.